src/lang/checkers/SanityChecker.ml

```diff
@@ -167,240 +166,6 @@ module ScillaSanityChecker
-          stmt_checker env sts
-        )
-    )
-
-  let library_checker elib env =
-    foldM ~f:(fun aenv lentry ->
-      match lentry with
-      | LibTyp _ -> pure aenv
-      | LibVar (x, _, e) ->
-        let%bind n = expr_checker e aenv in
-        pure @@ add_env aenv x n
-    ) ~init:env elib.lentries
-
-  let one_msg_checker (cmod : cmodule) rlibs elibs =
-    (* Bind folds to Many as they aren't much useful. *)
-    let env_rec = List.fold_left (fun acc le ->
-        match le with
-        | LibVar (i, _, _) -> add_env acc i Many
-        | LibTyp _ -> acc
-      ) (mk_env()) rlibs
-    in
-    (* First scan the external library functions. *)
-    let%bind env_elibs =
-      let rec recurser elibl =
-        foldM ~f:(fun acc_env elib ->
-          let%bind dep_env = recurser elib.deps in
-          let%bind lib_env = library_checker elib.libn dep_env in
-          (* Retain only env entries from elib. *)
-          let lib_env' = filter_env lib_env ~f:(fun name ->
-            List.exists (function | LibTyp _ -> false | LibVar (i, _, _) -> get_id i = name) elib.libn.lentries
-          ) in
-          let acc_env' = append_env acc_env lib_env' in
-          pure acc_env'
-        ) ~init:env_rec elibl
-      in
-      recurser elibs
-    in
-    (* Scan contract library if it exists. *)
-    let%bind env_libs =
-      match cmod.libs with
-      | Some lib ->
-        let%bind lib_env = library_checker lib env_elibs in
-        pure lib_env
-      | None -> pure env_elibs
-    in
-
-    (* Bind contract parameters to Many. *)
-    let env = List.fold_left (fun acc (p, _) ->
-        add_env acc p Many
-      ) env_libs (SCU.append_implict_contract_params cmod.contr.cparams)
-    in
-    (* Check all components. *)
-    let%bind _ = foldM ~f:(fun accenv cp ->
-        match cp.comp_type with
-        | CompTrans ->
-          (* Bind transition parameters to Many. *)
-          let params = SCU.append_implict_comp_params cp.comp_params in
-          let env' = List.fold_left (fun acc (p, _) ->
-              add_env acc p Many
-            ) accenv params
-          in
-          let%bind () = stmt_checker env' cp.comp_body in
-          pure accenv
-        | CompProc ->
-          let check_proc_clos = MArg (fun nmlist ->
-            let params' = fst @@ Core.List.unzip @@ SCU.append_implict_comp_params cp.comp_params in
-            let nmlist' = [Many; Many] @ nmlist in (* For the two appeneded implicit comp params above. *)
-            (* Analyze the body *)
-            let env' = add_env_list accenv params' nmlist' in
-            let%bind _ = stmt_checker env' cp.comp_body in
-            pure Many (* This doesn't matter because the procedure doesn't return a value. *)
-          ) in
-          (* bind check_proc_clos to this procedure. *)
-          pure @@ add_env accenv cp.comp_name check_proc_clos
-      ) ~init:env cmod.contr.ccomps
-    in
-    pure ()
```

Comment:
I bet this change felt really, really good! ;-)

Reply:
ha ha, it really did :)